src/quagga.js

```diff
@@ -195,14 +201,29 @@ function transformResult(result) {
     }
 }
 
+function addResult (result, imageData) {
+    var i;
+
+    if (!imageData || !result || !_resultCollector) {
+        return;
+    }
+
+    if (result.barcodes) {
+        for (i = 0; i < result.barcodes.length; i++) {
+            addResult(result.barcodes[i], imageData);
+        }
+        return;
+    }
+
+    if (result.codeResult) {
+        _resultCollector.addResult(imageData, _inputStream.getCanvasSize(), result.codeResult);
+    }
+}
+
 function publishResult(result, imageData) {
     if (_onUIThread) {
         transformResult(result);
-        if (imageData && result && result.codeResult) {
-            if (_resultCollector) {
-                _resultCollector.addResult(imageData, _inputStream.getCanvasSize(), result.codeResult);
-            }
-        }
+        addResult(result, imageData);
     }
 
     Events.publish("processed", result);
```

Comment:
The `detected` event is not fired any more when `multiple` is activated. Maybe you could expand the condition to include something like: `result.barcodes && result.barcodes.length > 0` . And since you only add the results to this array if the decoding was successful, this should be sufficient.